### PR TITLE
Breaking: do not require Babel configuration

### PR DESCRIPTION
**Pull request by kaicataldo**

### CODE REVIEW

File: lib/plugin.js [hunk 1]
```diff
@@ -1,112 +1,45 @@
...
-  if (typeof babelConfig !== "function") {
-    throw new Error("`babelConfig` is required and must be a function.");
-  }
+  const ssrBabelConfig = babelConfig.generate({
+    isBrowser: false,
+    targets,
+    config: userBabelConfig,
+  });
+  const browserBabelConfig = babelConfig.generate({
+    isClientBundle: true,
```
Comment (dpfavand) [inline lib/plugin.js:39]: `isClientBundle` isn't destructured in `generate()` - should this be `isBrowser`?

File: lib/plugin.js [hunk 2]
```diff
@@ -1,112 +1,45 @@
...
+    targets,
+    babelConfig: userBabelConfig,
```
Comment (dpfavand) [inline lib/plugin.js:41]: `babelConfig` isn't destructured in `generate()` - should be `config`?

**Review by dpfavand — COMMENTED** [review]
Overall, I think this provides a nice happy path and find this preferable to writing a full babelConfig in each project's `.eleventy.js`.

I'm concerned that there is no path for modifying the configurations for the default presets (you can only add additional presets and plugins), but I'm not sure if that is a common need. I _think_ that for non-happy-path scenarios, it might be preferable to use the callback pattern, since at that point I don't know what the user needs to customize? But there's a lot of stuff happening here to make sure modules are resolved that might make that trickier it appears?

**Comment by kaicataldo** [comment]
Alright, this has changed a bit - I'm going to go ahead and merge and publish this. I've decided to keep these as peer deps but to publish a separate preset that a user can add so that they don't have to install all the peer deps :)

**Pull request merged**